docs/ksm.md

```diff
@@ -0,0 +1,130 @@
+# Kernel Same-Page (KSM) tuning
+
+This section describes why `cc-proxy` tunes the host kernel
+KSM settings and how it does it.
+
+## What is KSM ?
+
+KSM is a host Linux kernel feature for de-duplicating memory pages.
+Although it was initially designed as a KVM specific feature, it is
+now part of the generic Linux memory management subsystem and can
+be leveraged by any userspace component or application looking for
+memory to save.
+
+A daemon (ksmd) periodically scans userspace memory, looking for
+identical pages that can be replaced by a single, write-protected
+page. When a process tries to modify this shared page content, it
+gets a private copy into its memory space. KSM only look at
+anonymous pages and does not go through the pagecache. Also, KSM
+only merges pages that have been explictly tagged as mergeable by
```

Comment:
I think KSM only scans (and hence potentially merges) pages that have been registered via madvise. The above implies the kernel might scan all anonymous pages and only merge them if they are marked via madvise.